python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1762,6 +1763,105 @@ def getURI(self, ref: DatasetRef, predict: bool = False) -> ResourcePath:
             )
         return primary
 
+    def getManyURIs(
+        self,
+        refs: Iterable[DatasetRef],
+        predict: bool = False,
+        throwIfNotFound: bool = True,
+    ) -> Dict[DatasetRef, DatasetRefURIs]:
+        # Docstring inherited
+
+        PrimaryType = Optional[ResourcePath]
+        ComponentsType = Dict[str, ResourcePath]
+        uris: Dict[DatasetRef, DatasetRefURIs] = {}
+
+        records = self._get_stored_records_associated_with_refs(refs)
+
+        missingRefs = [ref for ref in refs if ref.id not in records.keys()]
+
+        for ref in missingRefs:
+            # if this has never been written then we have to guess
+            if not self.exists(ref) and not predict:
```

Comment:
We know by definition that `self.exists(ref)` is False (because it will query the datastore records first and find that there is no entry) so this check tells us nothing. We can raise immediately `if missingRefs and not predict`.

The exists() call in getURIs is checking that the datastore knows about the file and that the file is there. getManyURIs is not checking that the file is really there, it's assuming it is if datastore thinks it is. We probably should document that distinction.